# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/301_struct_name_mismatch.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/301_struct_name_mismatch.rs
--- a/third_party/rust/ron/tests/301_struct_name_mismatch.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/301_struct_name_mismatch.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,4 +1,4 @@
-use ron::error::{Error, Position, SpannedError};
+use ron::error::{Error, Position, Span, SpannedError};
 use serde::Deserialize;
 
 #[derive(Debug, Deserialize, PartialEq)]
@@ -30,14 +30,20 @@
                 expected: "MyUnitStruct",
                 found: String::from("MyUnit")
             },
-            position: Position { line: 1, col: 7 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 7 }
+            }
         }),
     );
     assert_eq!(
         ron::from_str::<MyUnitStruct>("42"),
         Err(SpannedError {
             code: Error::ExpectedNamedStructLike("MyUnitStruct"),
-            position: Position { line: 1, col: 1 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 }
+            }
         }),
     );
 }
@@ -59,14 +65,20 @@
                 expected: "MyTupleStruct",
                 found: String::from("MyTypleStruct")
             },
-            position: Position { line: 1, col: 14 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 14 }
+            }
         }),
     );
     assert_eq!(
         ron::from_str::<MyTupleStruct>("42"),
         Err(SpannedError {
             code: Error::ExpectedNamedStructLike("MyTupleStruct"),
-            position: Position { line: 1, col: 1 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 }
+            }
         }),
     );
 }
@@ -88,14 +100,20 @@
                 expected: "MyNewtypeStruct",
                 found: String::from("MyNewtypeStrucl")
             },
-            position: Position { line: 1, col: 16 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 16 }
+            }
         }),
     );
     assert_eq!(
         ron::from_str::<MyNewtypeStruct>("42"),
         Err(SpannedError {
             code: Error::ExpectedNamedStructLike("MyNewtypeStruct"),
-            position: Position { line: 1, col: 1 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 }
+            }
         }),
     );
 }
@@ -117,14 +135,20 @@
                 expected: "MyStruct",
                 found: String::from("MuStryct")
             },
-            position: Position { line: 1, col: 9 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 9 }
+            }
         }),
     );
     assert_eq!(
         ron::from_str::<MyStruct>("42"),
         Err(SpannedError {
             code: Error::ExpectedNamedStructLike("MyStruct"),
-            position: Position { line: 1, col: 1 }
+            span: Span {
+                start: ron::error::Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 }
+            }
         }),
     );
 }